### DISCUSSION TITLE
ScaleIntensityRanged values

### DISCUSSION THREAD

**Discussion by talhaanwarch** [Project-MONAI/MONAI#2583] [Q&A] [answered]
In the [spleen tutorial](https://github.com/Project-MONAI/tutorials/blob/master/3d_segmentation/spleen_segmentation_3d_lightning.ipynb), how values of a_min and a_max  are selected for `ScaleIntensityRanged` ?
How the value of -57 and 164 are chosen? Are the specific for spleen data only or can be used for another dataset?
`ScaleIntensityRanged(keys=["image"], a_min=-57, a_max=164,b_min=0.0, b_max=1.0, clip=True),`
When I try `ScaleIntensityRanged` i got min as 0 and max as 1. 
But when I do `ScaleIntensityd` the max is 0.9 and min is 0. 
I want to know which one is better scaling option

**Comment by Nic-Ma** [comment, marked as answer]
Hi @talhaanwarch ,

Thanks for your interest here.
I think the parameters are specific for the spleen dataset.
@ericspod @rijobro @wyli please feel free to correct me if I misunderstood it.

Thanks.

**Reply by ericspod** [reply]
I'm afraid I don't know why the parameters are what they are. The images are CT so should be scaled for Hounsfield units (or really not scaled at all). I can't find specific reference to these parameters so having not written this notebook I don't have an origin for the numbers. The only other author not mentioned here is @yiheng-wang-nv who might know.

**Reply by wyli** [reply]
I think those are typical windowing settings for abdominal CT to cover soft tissue/fat https://radiopaedia.org/articles/windowing-ct the particular values are tuned according to the experiments.

**Reply by yiheng-wang-nv** [reply]
Hi @talhaanwarch , I don't how do these values come from, but take nnunet (https://arxiv.org/pdf/1809.10486.pdf) for reference, 5% and 95% percentiles of all intensity values (where the corresponding label > 0) for the training dataset (Spleen) are used and the values are: -41 and 176.
The values are specific for spleen.

**Reply by talhaanwarch** [reply]
thanks, a lot. it is clear. if you can paste the code used to calculate 5% and 95% percentile, it will be of great help.

**Reply by yiheng-wang-nv** [reply]
Hi, I think add the way to calculate a suitable scale range and spacing is useful, let me submit a PR for it.

**Reply by talhaanwarch** [reply]
@yiheng-wang-nv  please, mention, when you have submmitted it

**Reply by yiheng-wang-nv** [reply]
Hi @talhaanwarch , it has been implemented in: https://github.com/Project-MONAI/MONAI/pull/2616
You can take the unit test case for reference: 
https://github.com/Project-MONAI/MONAI/blob/dev/tests/test_dataset_summary.py

**Reply by talhaanwarch** [reply]
thanks a lot @yiheng-wang-nv . I got target_spacing as `(0.8115235, 0.8115235, 1.0)` should i set it as pixdim in Spacingd?
